contracts/schemes/JoinAndQuit.sol

```diff
@@ -0,0 +1,244 @@
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _decision a parameter of the voting result, 1 yes and 2 is no.
+    */
+    function executeProposal(bytes32 _proposalId, int256 _decision)
+    external
+    onlyVotingMachine(_proposalId)
+    returns(bool) {
+        require(proposals[_proposalId].accepted == false);
+        require(proposals[_proposalId].proposedMember != address(0));
+        Proposal memory proposal = proposals[_proposalId];
+        // Check if vote was successful:
+        if (_decision == 1) {
+            proposals[_proposalId].accepted = true;
+            address(fundingToken).safeTransfer(address(avatar), proposals[_proposalId].funding);
+            fundings[proposal.funder] = fundings[proposal.funder].add(proposal.funding);
+            totalDonation = totalDonation.add(proposal.funding);
+            checkFundedBeforeDeadLine();
+        } else {
+            address(fundingToken).safeTransfer(proposal.funder, proposal.funding);
+        }
+        emit ProposalExecuted(address(avatar), _proposalId, _decision);
+        return true;
+    }
+
+    /**
+    * @dev Submit a proposal for to join in a dao
+    * @param _descriptionHash A hash of the proposal's description
+    * @param _feeAmount - the amount to fund the dao with. should be >= the minimum fee to join
+    * @param _member the member to join in - if this address is zero the msg.sender will be set as the member
+    * @return proposalId the proposal id
+    */
+    function proposeToJoinIn(
+        string memory _descriptionHash,
+        uint256 _feeAmount,
+        address _member
+    )
+    public
+    returns(bytes32)
+    {
+        require(_feeAmount >= minFeeToJoin, "_feeAmount should be >= then the minFeeToJoin");
+        address(fundingToken).safeTransferFrom(msg.sender, address(this), _feeAmount);
+        bytes32 proposalId = votingMachine.propose(2, voteParams, msg.sender, address(avatar));
+        address member = _member;
+        if (member == address(0)) {
+            member = msg.sender;
+        }
+
+        Proposal memory proposal = Proposal({
+            accepted: false,
+            proposedMember: member,
+            funding : _feeAmount,
+            funder : msg.sender
+        });
+        proposals[proposalId] = proposal;
+
+        emit JoinInProposal(
+            address(avatar),
+            proposalId,
+            _descriptionHash,
+            member,
+            _feeAmount
+        );
+
+        proposalsInfo[address(votingMachine)][proposalId] = ProposalInfo({
+            blockNumber:block.number,
+            avatar:avatar
+        });
+        return proposalId;
+    }
+
+    /**
+    * @dev RedeemReputation reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @return reputation the redeemed reputation.
+    */
+    function redeemReputation(bytes32 _proposalId) public returns(uint256 reputation) {
```

Comment:
Must this be a separate function? What about making this a part of the `executeProposal` function?

Reply:
we can do that though :
1. the `withdraw pattern` is more secure. https://ethereum.stackexchange.com/questions/67342/what-is-the-withdrawal-pattern 
2. do we want the last voter for the case of abs vote or the executer will pay the gas ? 

so why not sending the funds on the `withdraw pattern` ? good question .we might need that  that immediately upon the proposal execution /dao decision.